staging/src/k8s.io/client-go/rest/request_test.go

```diff
@@ -341,20 +341,51 @@ func TestResultIntoWithNoBodyReturnsErr(t *testing.T) {
 
 func TestURLTemplate(t *testing.T) {
 	uri, _ := url.Parse("http://localhost")
-	r := NewRequest(nil, "POST", uri, "", ContentConfig{GroupVersion: &schema.GroupVersion{Group: "test"}}, Serializers{}, nil, nil, 0)
-	r.Prefix("pre1").Resource("r1").Namespace("ns").Name("nm").Param("p0", "v0")
-	full := r.URL()
-	if full.String() != "http://localhost/pre1/namespaces/ns/r1/nm?p0=v0" {
-		t.Errorf("unexpected initial URL: %s", full)
-	}
-	actualURL := r.finalURLTemplate()
-	actual := actualURL.String()
-	expected := "http://localhost/pre1/namespaces/%7Bnamespace%7D/r1/%7Bname%7D?p0=%7Bvalue%7D"
-	if actual != expected {
-		t.Errorf("unexpected URL template: %s %s", actual, expected)
-	}
-	if r.URL().String() != full.String() {
-		t.Errorf("creating URL template changed request: %s -> %s", full.String(), r.URL().String())
+	testCases := []struct {
+		Request          *Request
+		ExpectedFullURL  string
+		ExpectedFinalURL string
+	}{
+
+		{
+			Request: NewRequest(nil, "POST", uri, "", ContentConfig{GroupVersion: &schema.GroupVersion{Group: "test"}}, Serializers{}, nil, nil, 0).
+				Prefix("pre1").Resource("r1").Namespace("ns").Name("nm").Param("p0", "v0"),
+			ExpectedFullURL:  "http://localhost/pre1/namespaces/ns/r1/nm?p0=v0",
+			ExpectedFinalURL: "http://localhost/pre1/namespaces/%7Bnamespace%7D/r1/%7Bname%7D?p0=%7Bvalue%7D",
+		},
+		{
+			Request: NewRequest(nil, "DELETE", uri, "", ContentConfig{GroupVersion: &schema.GroupVersion{Group: "test"}}, Serializers{}, nil, nil, 0).
+				Prefix("/api/v1/namespaces/ns/r1"),
+			ExpectedFullURL:  "http://localhost/api/v1/namespaces/ns/r1",
+			ExpectedFinalURL: "http://localhost/api/v1/namespaces/%7Bnamespace%7D/r1",
+		},
+		{
+			Request: NewRequest(nil, "DELETE", uri, "", ContentConfig{GroupVersion: &schema.GroupVersion{Group: "test"}}, Serializers{}, nil, nil, 0).
+				Prefix("/apis/apps/v1/namespaces/ns/r1"),
+			ExpectedFullURL:  "http://localhost/apis/apps/v1/namespaces/ns/r1",
+			ExpectedFinalURL: "http://localhost/apis/apps/v1/namespaces/%7Bnamespace%7D/r1",
+		},
+		{
+			Request: NewRequest(nil, "DELETE", uri, "", ContentConfig{GroupVersion: &schema.GroupVersion{Group: "test"}}, Serializers{}, nil, nil, 0).
+				Prefix("/apis/apps/v1/namespaces"),
+			ExpectedFullURL:  "http://localhost/apis/apps/v1/namespaces",
+			ExpectedFinalURL: "http://localhost/apis/apps/v1/namespaces",
+		},
```

Comment:
Please also have a test case where there is some prefix before api or apis.

Reply:
By looking at https://github.com/kubernetes/kubernetes/blob/45c5ddd2fb8f10e097483257663fa6a7be59ffa9/staging/src/k8s.io/client-go/dynamic/simple.go#L307, it doesn't seems like there is possibilities of having anything other than api and apis in r.pathPrefix, do you mean have a r.baseURL.Path because of https://github.com/kubernetes/kubernetes/blob/426ef9d349bb3a277c3e8826fb772b6bdb008382/staging/src/k8s.io/client-go/rest/request.go#L275?

Reply:
Yes abspath is one way to get it, you can also get it by monkeying with the config, I think.

Reply:
Thanks.